java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/preparation/IndexRouteCreator.java

```diff
@@ -228,23 +228,25 @@ private QuadRect flipBbox(QuadRect bbox) {
 		return new QuadRect(bbox.left, bbox.bottom, bbox.right, bbox.top);
 	}
 	
-	private void addLowEmissionZoneTag(Way e) {
+	private Map<String, String> addLowEmissionZoneTag(Way e, Map<String, String> tags) {
```

Comment:
Return value or change by link, need select single

Reply:
This pattern of usage 
```
tags = tagsTransformer.addPropogatedTags(renderingTypes, EntityConvertApplyType.ROUTING, e, tags);
			tags = renderingTypes.transformTags(tags, EntityType.WAY, EntityConvertApplyType.ROUTING);
```
It's optimized not to create copy when not needed